alc/backends/coreaudio.cpp

```diff
@@ -552,6 +593,44 @@ bool CoreAudioPlayback::reset()
             err);
         return false;
     }
+    
+    if(!mDevice->Flags.test(ChannelsRequest))
+    {
+        UInt32 propSize{};
+        Boolean writable;
+        err = AudioUnitGetPropertyInfo(mAudioUnit,
+                                       kAudioUnitProperty_AudioChannelLayout,
+                                       kAudioUnitScope_Output,
+                                       OutputElement,
+                                       &propSize,
+                                       &writable);
+        AudioChannelLayout *layout = new AudioChannelLayout[propSize]();
```

Comment:
This should also use `std::make_unique` or `std::vector` to avoid leaking the memory, and needs to use a raw bytes to reinterpret as the layout struct, like how `GetDeviceChannelCount` does it.